src/components/Wizard/CreateVmWizard/providers/VCenterVmsWithPrefill.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+
+import { Dropdown } from '../../../Form';
+import { settingsValue } from '../../../../k8s/selectors';
+
+import { BATCH_CHANGES_KEY, DESCRIPTION_KEY, NAME_KEY, PROVIDER_VMWARE_VM_KEY } from '../constants';
+
+class VCenterVmsWithPrefill extends React.Component {
+  state = {
+    lastName: undefined, // last prefilled VM name value
+    lastDescription: undefined,
+  };
+
+  prefillVmName(basicSettings, onFormChange, vmVmware) {
+    const value = get(vmVmware, ['Config', 'Name']);
+    const formName = settingsValue(basicSettings, NAME_KEY);
+    if (!formName || formName === this.state.lastName) {
+      if (this.state.lastName !== value) {
+        // avoid infinite loop
+        this.setState({ lastName: value });
+        return { value, target: NAME_KEY };
+      }
+    }
+    return undefined;
+  }
+
+  prefillVmDescription(basicSettings, onFormChange, vmVmware) {
+    const value = get(vmVmware, ['Config', 'Annotation']);
+    const formValue = settingsValue(basicSettings, DESCRIPTION_KEY);
+    if (!formValue || formValue === this.state.lastDescription) {
+      if (this.state.lastDescription !== value) {
+        // avoid infinite loop
+        this.setState({ lastDescription: value });
+        return { value, target: DESCRIPTION_KEY };
+      }
+    }
+    return undefined;
+  }
+
+  prefillValues(basicSettings, onFormChange, vmVmware) {
+    const result = [];
+
+    const namePair = this.prefillVmName(basicSettings, onFormChange, vmVmware);
+    if (namePair) {
+      result.push(namePair);
+    }
+
+    const descrPair = this.prefillVmDescription(basicSettings, onFormChange, vmVmware);
+    if (descrPair) {
+      result.push(descrPair);
+    }
+
+    if (result.length > 0) {
+      onFormChange({ value: result }, BATCH_CHANGES_KEY);
```

Comment:
since you are calling onFormChange, not onChange, you need to also validate field.  See https://github.com/kubevirt/web-ui-components/blob/master/src/components/Form/FormFactory.js#L149 149-161. Maybe you could extract it to separate function and reuse it here

Reply:
done